indico/modules/rb_new/util.py

```diff
@@ -154,34 +160,48 @@ def get_existing_room_occurrences(room, start_dt, end_dt, allow_overlapping=Fals
         query = query.filter(ReservationOccurrence.start_dt >= start_dt, ReservationOccurrence.end_dt <= end_dt)
     if only_accepted:
         query = query.filter(Reservation.is_accepted)
-    return query.all()
-
+    return group_list(query, key=lambda obj: obj.reservation.room.id)
 
-def get_room_conflicts(room, start_dt, end_dt, repeat_frequency, repeat_interval):
-    conflicts = []
-    pre_conflicts = []
 
+def get_rooms_conflicts(rooms, start_dt, end_dt, repeat_frequency, repeat_interval):
+    room_conflicts = {}
     candidates = ReservationOccurrence.create_series(start_dt, end_dt, (repeat_frequency, repeat_interval))
-    occurrences = ReservationOccurrence.find_overlapping_with(room, candidates).all()
+    room_ids = map(attrgetter('id'), rooms)
+    query = (ReservationOccurrence
+             .find(Reservation.room_id.in_(room_ids),
```

Comment:
we don't use `.find()` in new code; use `.query` and `.filter()` instead

Reply:
:confused: Really?